skbio/sequence/_nucleotide_mixin.py

```diff
@@ -161,6 +161,15 @@ def complement(self, reverse=False):
 
         if reverse:
             complement = complement[::-1]
+
+        if self.has_interval_metadata():
+            interval_metadata = self.interval_metadata
```

Comment:
I will fix it. but your code won't work, because `complement = complement[::-1]` drops `interval_metadata`, making the following line never executed.